### ISSUE TITLE
[Bug] Memory Leak when Using Image in Certain Containers 

### ISSUE THREAD

**Issue by LGE-AUGMENSYS** [xamarin/Xamarin.Forms#12733]
### Description
When using the Xamarin.Forms Image control to display font glyphs inside a Syncfusion tab view, the entire pages are kept in memory, causing memory leaks.
* Without the Image control, the leaks are gone;
* Without the container tab view, the leaks are gone as well in the attached example. However, there are still leaks in our real application. So I'm assuming it also leaks in certain other containers;
* There are no leaks when downgrading the Xamarin.Forms version to 4.6.

### Steps to Reproduce

1. Launch the attached example app in Visual Studio using the "Memory Usage" performance profiling ;
2. Click the "Leak Page" button;
3. Navigate back to the main page;
4. Gather a snapshot after some GC;
5. Repeat step 2 & 3 several times (5 in the example screenshots);
6. Repeat step 4 and make a diff between the two snapshots.

### Expected Behavior
No delta in the diff of live instances or at least no additional instances of the LeakPage and its associated instances.

### Actual Behavior
Many new instances are kept in memory and cannot be GC'ed. See screenshots below.

### Basic Information

- Version with issue: Xamarin.Forms 4.8.0.1534, 4.8.0.1560, 4.7.0.1351, probably more adjacent minor versions.
- Last known good version: Xamarin.Forms 4.6.0.1180
- IDE: Visual Studio 16.7.6
- Platform Target Frameworks:
  - iOS:  haven't tried
  - Android: 10.0 (tried in our real app, not in the attached example)
  - UWP:  19041.508
- Android Support Library Version: Xamarin.AndroidX.Core 1.3.0.1
- Nuget Packages: Syncfusion.Xamarin.SfTabView 18.3.0.44
- Affected Devices: Windows PC, Samsung Galaxy S10

### Screenshots

![leak_summary](https://user-images.githubusercontent.com/51995996/98095083-66908700-1e8a-11eb-8baf-0f1a8c6324f8.png)
![leak_details](https://user-images.githubusercontent.com/51995996/98095102-6bedd180-1e8a-11eb-85d4-a0c45ecf3e82.png)

### Reproduction Link

[FormsMemoryLeak.zip](https://github.com/xamarin/Xamarin.Forms/files/5486964/FormsMemoryLeak.zip)

### Workaround

Downgrading Xamarin.Forms. However, our product release would be heavily affected by this.

**Comment by rmarinho** [comment]
Does it work fine if you are not using Syncfusion tab view ?

**Comment by LGE-AUGMENSYS** [comment]
No. In our real application, it still leaks even if I remove the Syncfusion tab view. So I assume it has the same problem in some other containers. Due to our time pressure, I've only managed to recreate the issue using the tab view in the attached example.

**Comment by LGE-AUGMENSYS** [comment]
We found out that if we explicitly call Dispose on the Image control, the leaks are gone.
However, other controls seem to have the same problem. So it looks like certain containers have problems propagating the disposal?
Or did the disposal mechanism (or the way view hierarchical tree is managed) change after Xamarin.Forms 4.6?

**Comment by hartez** [comment]
> We found out that if we explicitly call Dispose on the Image control, the leaks are gone.
> However, other controls seem to have the same problem. So it looks like certain containers have problems propagating the disposal?

What other "containers" are you using? 

> No. In our real application, it still leaks even if I remove the Syncfusion tab view. So I assume it has the same problem in some other containers.

If I remove the tab view and just have the Image as the content of the Page, there is no leak. So your theory that it's container-dependent seems sound, but right now the only example we have is the Syncfusion Tab. So unless you can tell us of another type of container which causes this issue, I think this is a Syncfusion bug, not a Forms bug.

**Comment by LGE-AUGMENSYS** [comment]
> 
> 
> > We found out that if we explicitly call Dispose on the Image control, the leaks are gone.
> > However, other controls seem to have the same problem. So it looks like certain containers have problems propagating the disposal?
> 
> What other "containers" are you using?
> 
> > No. In our real application, it still leaks even if I remove the Syncfusion tab view. So I assume it has the same problem in some other containers.
> 
> If I remove the tab view and just have the Image as the content of the Page, there is no leak. So your theory that it's container-dependent seems sound, but right now the only example we have is the Syncfusion Tab. So unless you can tell us of another type of container which causes this issue, I think this is a Syncfusion bug, not a Forms bug.

Other "containers" include mostly ContentView with dynamically loaded templates, Syncfusion list view, pull to refresh. I provided the example to Syncfusion and they claim to have fixed it (just the tab view, though). I am still waiting to see the fix.

However, one thing is also clear that the same setup (our app, Syncfusion controls) has no leaks in Forms 4.6 (details in the Basic Information section). Of course it could still be that some basic Forms concepts/APIs were misunderstood/misused and are now broken "as expected". I raised the same question to Syncfusion but unfortunately got no help.

I know it's a difficult ask, but I would really appreciate if you guys having more insights on the Forms version related changes could offer some hints or directions of thoughts.

**Comment by LGE-AUGMENSYS** [comment]
So I finally got some free time to further compare the differences between our leaking app and the example.
It turns out the leaks are only present in a Master-Detail page setup. I'm linking the modified example app below.
[FormsMemoryLeak.zip](https://github.com/xamarin/Xamarin.Forms/files/5653285/FormsMemoryLeak.zip)

My testing is done using Visual Studio's builtin profiler. From the start page of the app:

1. Navigate to the TitlePage ("Title Leak") and then back;
2. Several explicit GCs, wait a bit and take a snapshot;
4. Repeat step 1 five times;
5. Several explicit GCs, wait a bit and take a snapshot;
7. Diff those snapshots.

![Snipaste_2020-12-07_15-45-38](https://user-images.githubusercontent.com/51995996/101367395-65e27a80-38a6-11eb-9330-134eefef42d8.png)
![Snipaste_2020-12-07_15-45-45](https://user-images.githubusercontent.com/51995996/101367412-6b3fc500-38a6-11eb-8baa-94f6708d9976.png)

You can see there are exactly five additional instances of the TitlePage remaining.
To summarize:

- If I change the code in FormsMemoryLeak.App and do not use a MasterDetailPage, the leaks are gone;
- If I remove the Image control from the title area (and I suspect it has a similar leaking pattern when placed inside other containers), the leaks are gone;
- And Xamarin.Forms 4.6 doesn't seem to have this leak problem.

I hope this helps.
